src/control/system/database.go

```diff
@@ -468,27 +468,29 @@ func (db *Database) GroupMap() (*GroupMap, error) {
 	return gm, nil
 }
 
-// ReplicaRanks returns the set of ranks associated with MS replicas.
-func (db *Database) ReplicaRanks() (*GroupMap, error) {
+// GroupMapWithReplicaRanks returns the latest system group map with
+// the set of ranks associated with MS replicas.
+func (db *Database) GroupMapWithReplicaRanks() (*GroupMap, []Rank, error) {
```

Comment:
Given that we are moving toward a future where the entire GroupMap is always sent to the client, I think I would prefer to consolidate the `ReplicaRanks()` and `GroupMap()` methods, and extend the `GroupMap` type to hold the `MsRanks` slice. So with this proposal, we would have a single `GroupMap()` method that returns a `(*GroupMap, error)` and it always contains the slice of MS ranks. The overhead of generating the MS ranks slice is negligible.

Reply:
Sure, I'll see what I can do...